freqtrade/strategy/interface.py

```diff
@@ -633,6 +651,42 @@ def should_sell(self, trade: Trade, rate: float, date: datetime, buy: bool,
         # logger.debug(f"{trade.pair} - No sell signal.")
         return SellCheckTuple(sell_type=SellType.NONE)
 
+    def entry_price_reached(self, pair: str, current_rate: float,
+                          current_time: datetime, low: float = None,
+                          high: float = None, side: str = "long") -> bool:
+        """
+        Based on current candle low ,decides if entry price was reached
+        :param current_rate: current rate
+        :param low: Low value of this candle, only set in backtesting
+        :param high: High value of this candle, only set in backtesting
+        """
+
+        if self.use_custom_entry_price:
+            entry_price_value = strategy_safe_wrapper(self.custom_entry_price, default_retval=None
+                                                    )(pair=pair,
+                                                      current_time=current_time,
+                                                      current_rate=current_rate)
```

Comment:
I'm not sure we need this whole function.
Basically - you should move the highlighted code to `execute_buy` (right after `get_rate()`) - and the same to backtesting.py.

The remainder of the logic is only relevant for backtesting (and i'd not add short logic at this point in time - thats something we'll add at a later point only).
However, i'm not quite sure simulating failed entries will make sense to backtesting - as usually, your offsets should be rather tight - so the only case when entries fail is if you're "outside" / below the candle.
That won't however properly simulate "real" trades - as even prices within the candle may never be reached again.

Reply:
So I remove logs and short logics,  However I think It's useful to simulate failed entries. The distance between current price and your proposed entry price can be an hyperparameter for hyperopt

Reply:
At the moment, neither of the 2 added methods is called from backtesting - so it's not simulating anything.

In general, i'd agree - assuming you find a way to reliably and realistically simulate failed entries.
simply "below low" will however not do it in my opinion (even though it might be an acceptable start).


In live, you have to assume that an entry happens 2-5s after open (depending on the length of the pairlist and complexity of a strategy, it can also be more). You'll not know if (by that time) low already happened or not.
Sure, on longer timeframes (1h and up) - chances are pretty high that low is still going to happen.

Especially on smaller timesframes (1m, 5m) - you can however encounter cases where low is hit "immediately" before taking off - so even though your order is within the candle, it'll never be reached.

I'm fine with it as long as we clearly document the shortcommings of this function - otherwise it'll boost assumptions on the reliability of backtesting which are simply not there.

---

As mentioned above - as this is a function only relevant to backtesting, please move it to the backtesting file, having it in `interface.py` will be missleading (and for dry-run, we have a better way to simulate failed entries onw).